### PR TITLE
Fix Cosine with Warmup Scheduler

### PR DESCRIPTION
**Pull request by w11wo**
I've been training LLMs with EasyDeL and noticed that my learning rates aren't warming up and decaying properly when using the Cosine with Linear Warmup scheduler (it is always warming up and never decaying). Looks like this should fix it.

Reference: [Optax Docs](https://optax.readthedocs.io/en/latest/api/optimizer_schedules.html#optax.warmup_cosine_decay_schedule)

### CODE REVIEW

**Comment by erfanzar** [comment]
Thanks for contributing to FJFormer❤️
